ui/app/AppLayouts/Wallet/views/FollowingAddresses.qml

```diff
@@ -0,0 +1,199 @@
+
+        function performSearch() {
+            var offset = (currentPage - 1) * pageSize
+            isPaginationLoading = true
+            RootStore.refreshFollowingAddresses(currentSearch, pageSize, offset)
+        }
+
+        function goToPage(pageNumber) {
+            currentPage = pageNumber
+            performSearch()
+        }
+
+        function refresh() {
+            currentPage = 1
+            currentSearch = ""
+            searchBox.text = ""
+            isPaginationLoading = true
+            RootStore.refreshFollowingAddresses("", pageSize, 0)
+        }
+    }
+
+    Connections {
+        target: walletSectionFollowingAddresses
+        function onFollowingAddressesUpdated() {
+            d.isPaginationLoading = false
+        }
+    }
+
+    Connections {
+        target: RootStore.followingAddresses
+        function onCountChanged() {
+            d.isPaginationLoading = false
+        }
+    }
+
+    Component.onCompleted: {
+        d.refresh()  // Load data when user navigates to this page
+    }
+
+    Timer {
+        id: searchDebounceTimer
+        interval: 250
+        repeat: false
+        onTriggered: d.performSearch()
+    }
+
+    ColumnLayout {
+        anchors.fill: parent
+        spacing: 0
+
+        SearchBox {
+            id: searchBox
+            Layout.fillWidth: true
+            Layout.bottomMargin: Theme.padding
+            visible: true
+            placeholderText: qsTr("Search for name, ENS or address")
+
+            onTextChanged: {
+                d.currentSearch = text
+                d.currentPage = 1
+                searchDebounceTimer.restart()
+            }
+
+            validators: [
+                StatusValidator {
+                    property bool isEmoji: false
+
+                    name: "check-for-no-emojis"
+                    validate: (value) => {
+                                  if (!value) {
+                                      return true
+                                  }
+
+                                  isEmoji = Constants.regularExpressions.emoji.test(value)
+                                  if (isEmoji){
+                                      return false
+                                  }
+
+                                  return Constants.regularExpressions.alphanumericalExpanded1.test(value)
+                              }
+                    errorMessage: isEmoji?
+                                      qsTr("Your search is too cool (use A-Z and 0-9, single whitespace, hyphens and underscores only)")
+                                    : qsTr("Your search contains invalid characters (use A-Z and 0-9, single whitespace, hyphens and underscores only)")
+                }
+            ]
+        }
+
+        ShapeRectangle {
+            id: noFollowingAddresses
+            Layout.fillWidth: true
+            Layout.preferredHeight: 44
+            visible: RootStore.followingAddresses.count === 0 && !d.isPaginationLoading
```

Comment:
On a second thought, do not use the singleton `RootStore` inside this component at all; you can pass it down from the parent component (you already have it there as `property WalletStores.RootStore rootStore`), or better yet, just use plain properties and signals